tests/datastore_redis/test_instance_info.py

```diff
@@ -45,95 +44,96 @@ class DisabledSSLConnection(redis.SSLConnection, DisabledConnection):
 
-@pytest.mark.parametrize('args,kwargs,expected', _instance_info_tests)
+
+@pytest.mark.parametrize("args,kwargs,expected", _instance_info_tests)
 def test_redis_connection_instance_info(args, kwargs, expected):
     r = redis.Redis(*args, **kwargs)
     r.connection_pool.connection_class = DisabledConnection
-    connection = r.connection_pool.get_connection('SELECT')
+    connection = r.connection_pool.get_connection("SELECT")
     try:
         conn_kwargs = _conn_attrs_to_dict(connection)
         assert _instance_info(conn_kwargs) == expected
     finally:
         r.connection_pool.release(connection)
 
-@pytest.mark.parametrize('args,kwargs,expected', _instance_info_tests)
+
+@pytest.mark.parametrize("args,kwargs,expected", _instance_info_tests)
 def test_strict_redis_connection_instance_info(args, kwargs, expected):
     r = redis.StrictRedis(*args, **kwargs)
     r.connection_pool.connection_class = DisabledConnection
-    connection = r.connection_pool.get_connection('SELECT')
+    connection = r.connection_pool.get_connection("SELECT")
     try:
         conn_kwargs = _conn_attrs_to_dict(connection)
         assert _instance_info(conn_kwargs) == expected
     finally:
         r.connection_pool.release(connection)
 
+
 _instance_info_from_url_tests = [
-    (('redis://localhost:1234/',), {}, ('localhost', '1234', '0')),
-    (('redis://localhost:1234',), {}, ('localhost', '1234', '0')),
-    (('redis://user:password@localhost:6379',), {}, ('localhost', '6379', '0')),
-    (('redis://localhost:6379/2',), {}, ('localhost', '6379', '2')),
-    (('redis://localhost:6379',), {'db': 2}, ('localhost', '6379', '2')),
-    (('redis://@127.0.0.1:6379',), {}, ('127.0.0.1', '6379', '0')),
-    (('redis://:1234/',), {}, ('localhost', '1234', '0')),
-    (('redis://@:1234/',), {}, ('localhost', '1234', '0')),
-    (('redis://localhost:1234/garbage',), {}, ('localhost', '1234', '0')),
+    (("redis://localhost:1234/",), {}, ("localhost", "1234", "0")),
+    (("redis://localhost:1234",), {}, ("localhost", "1234", "0")),
+    (("redis://user:password@localhost:6379",), {}, ("localhost", "6379", "0")),
+    (("redis://localhost:6379/2",), {}, ("localhost", "6379", "2")),
+    (("redis://localhost:6379",), {"db": 2}, ("localhost", "6379", "2")),
+    (("redis://@127.0.0.1:6379",), {}, ("127.0.0.1", "6379", "0")),
+    (("redis://:1234/",), {}, ("localhost", "1234", "0")),
+    (("redis://@:1234/",), {}, ("localhost", "1234", "0")),
+    (("redis://localhost:1234/garbage",), {}, ("localhost", "1234", "0")),
 ]
 
-if REDIS_PY_VERSION >= (2, 7, 5):
-    _instance_info_from_url_tests.append(
-        (('redis://127.0.0.1',), {}, ('127.0.0.1', '6379', '0'))
-    )
+# Behavior to default port to 6379 was added in v2.7.5
+# (https://github.com/redis/redis-py/commit/cfe1041bbb8a8887531810429879bffbe705b03a)
+# and removed in v4.0.0b1 (https://github.com/redis/redis-py/blame/v4.0.0b1/redis/connection.py#L997)
+if (3, 5, 3) >= REDIS_PY_VERSION >= (2, 7, 5):
```

Comment:
I don't think this is how this operator works. `(3, 5, 3) >= REDIS_PY_VERSION` evaluates to True or False, which gets coerced to 1 or 0 before being compared to `>= (2, 7, 5)`. So the full expression will always be False.

```suggestion
if (3, 5, 3) >= REDIS_PY_VERSION and REDIS_PY_VERSION >= (2, 7, 5):
```

Reply:
Ironically I just found this in another codebase, this might actually work. I'm not quite sure why it works, I didn't expect it to.

Reply:
Fascinating. It's actually baked into the syntax of the language. I retract that suggestion.

https://docs.python.org/3/reference/expressions.html#comparisons